planner/core/logical_plan_builder.go

```diff
@@ -3101,6 +3101,17 @@ func (b *PlanBuilder) buildUpdateLists(
 	return newList, p, allAssignmentsAreConstant, nil
 }
 
+// extractDefaultExpr extract a `DefaultExpr` without any parameter from a `ExprNode`
+// return it and if successful extract it
+// Note the sql function `DEFAULT(a)` is not same with keyword `DEFAULT`
+// Sql function `DEFAULT(a)` will return `false`
+func extractDefaultExpr(node ast.ExprNode) (*ast.DefaultExpr, bool) {
```

Comment:
Can we only return one value, which is the `expr`? the second bool value can be derived by `expr == nil`

Reply:
I think this way is easier to understand:
```go
if expr, ok := extractDefaultExpr(assign.Expr); ok {
	expr.Name = assign.Column
}
```
vs
```go
if expr := extractDefaultExpr(assign.Expr); expr != nil {
	expr.Name = assign.Column
}
```